Replace boolean gender field in interface tutorial (#3533)

playertr opened the pull request:
This addresses #3533.

playertr committed:
Replace boolean gender field in interface tutorial (#3533)

cdhabecker commented on source/Tutorials/Beginner-Client-Libraries/Single-Package-Define-And-Use-Interface.rst:
```diff
@@ -60,22 +60,21 @@ Inside ``more_interfaces/msg``, create a new file ``AddressBook.msg``, and paste
+   string phone_number
+   uint8 phone_type
 
 This message is composed of 5 fields:
```
Whoops, now it's 4. The sentence should not have specified a number in the first place. 

"This message is composed of these fields:" works for any number.

playertr replied:
Oops! Missed that. I can make that change when you're done reviewing or you can if you prefer.

cdhabecker replied:
I'm still working on my first pull request. I'm not ready to try to modify yours. :-)

playertr committed:
Change '5 fields' -> 'these fields'

fujitatomoya commented on source/Tutorials/Beginner-Client-Libraries/Single-Package-Define-And-Use-Interface.rst:
```diff
@@ -60,22 +60,21 @@ Inside ``more_interfaces/msg``, create a new file ``AddressBook.msg``, and paste
-* gender: of type bool, that can be either MALE or FEMALE
-* age: of type uint8
-* address: of type string
+* phone_number: of type uint32
```
```suggestion
* phone_number: of type string
```

playertr replied:
Oops, I think that the place to change this would be Line 69. We would want to change the message definition of phone_numbe to uint32. If it is a string, then the integer field assignments would not compile.

fujitatomoya commented on source/Tutorials/Beginner-Client-Libraries/Single-Package-Define-And-Use-Interface.rst:
```diff
@@ -194,9 +193,8 @@ In ``more_interfaces/src`` create a file called ``publish_address_book.cpp`` and
-          message.age = 30;
-          message.gender = message.MALE;
-          message.address = "unknown";
+          message.phone_number = 2068675309;
```
`2068675309` means anything specific?

```suggestion
          message.phone_number = 1234567890;
```

playertr replied:
Not really. I agree with the change to 1234567890.

playertr committed:
Update source/Tutorials/Beginner-Client-Libraries/Single-Package-Define-And-Use-Interface.rst

Changes example phone number from famous "8675309" with a Seattle area code to a more geographically neutral series

Co-authored-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

playertr committed:
Change phone_number field to uint32 instead of string

playertr committed:
Update John Doe's phone number in array example for consistency

playertr committed:
Change phone_number to uint64 to accommodate all possible 10-digit integers

playertr committed:
Update step-by-step code walkthrough to be consistent with initial block.

playertr commented:
I pushed a few more commits to address your feedback. Compared with the initial PR, it now:
- omits the erroneous statement "This message is composed of 5 fields"
- changes all instances of John Doe's phone number from 2068675309 to 1234567890.
- updates the step-by-step code walkthrough to match the initial example block
- alters the definition of `phone_number` to be of type `uint64`, which is large enough to store all 10-digit phone numbers (uint32 maxes out at 4,294,967,295)

Please pardon there errors. I now tested out the walkthrough code in a VM and it worked.

fujitatomoya commented on source/Tutorials/Beginner-Client-Libraries/Single-Package-Define-And-Use-Interface.rst:
```diff
@@ -60,22 +60,21 @@ Inside ``more_interfaces/msg``, create a new file ``AddressBook.msg``, and paste
-   bool gender
-   uint8 age
-   string address
+   uint64 phone_number
```
in semantic, phone number is supposed to be just a string information. for me it does not mean anything as integer information. probably we can keep this open for more feedback!

playertr replied:
Ah, understood. So a good way to handle this would be adding string quotes to the C++ assignment, like so?

```
# in AddressBook.msg:
...
string phone_number
...
```

```
// in publish_address_book.cpp
message.phone_number = "1234567890";
```

clalancette replied:
Yeah, agreed that this is better as a string.  Exactly what you put above @playertr should work fine.

playertr replied:
Done.

fujitatomoya reviewed (commented):
one minor comment needs more feedback, overall lgtm. thanks 👍

playertr committed:
Change phone numbers to strings

clalancette reviewed (approved):
This looks good to me, thanks for the fixes.

clalancette commented:
I'm going to go ahead and merge this, because @fujitatomoya 's last comment has been addressed.
